Dockerfile

```diff
@@ -1,46 +1,42 @@
-FROM redis AS builder
+ARG OS=debian:buster
 
-ENV DEPS "build-essential git ca-certificates curl unzip wget libgomp1 patchelf"
+#----------------------------------------------------------------------------------------------
+FROM redis AS redis
+FROM ${OS} AS builder
 
-# install latest cmake
-ADD https://cmake.org/files/v3.12/cmake-3.12.4-Linux-x86_64.sh /cmake-3.12.4-Linux-x86_64.sh
-RUN mkdir /opt/cmake
-RUN sh /cmake-3.12.4-Linux-x86_64.sh --prefix=/opt/cmake --skip-license
-RUN ln -s /opt/cmake/bin/cmake /usr/local/bin/cmake
-RUN cmake --version
+ARG PACK=0
+ARG TEST=0
 
-# Set up a build environment
-RUN set -ex;\
-    deps="$DEPS";\
-    apt-get update;\
-    apt-get install -y --no-install-recommends $deps
-
-# Get the dependencies
 WORKDIR /redisai
+COPY --from=redis /usr/local/ /usr/local/
+
+COPY ./deps/readies/ deps/readies/
+COPY ./system-setup.py .
+COPY ./test/test_requirements.txt test/
+
+RUN ./deps/readies/bin/getpy
+RUN ./system-setup.py
+
+COPY ./get_deps.sh .
+RUN ./get_deps.sh cpu
+
 ADD ./ /redisai
-RUN set -ex;\
-    mkdir -p deps;\
-    DEPS_DIRECTORY=deps bash ./get_deps.sh cpu
-
-# Build the source
-RUN set -ex;\
-    rm -rf build;\
-    mkdir -p build;\
-    cd build;\
-    cmake -DDEPS_PATH=../deps/install ..;\
-    make && make install;\
-    cd ..
-
-# Package the runner
+RUN make all
+RUN if [ "$PACK" = "1" ]; then make pack; fi
+RUN if [ "$TEST" = "1" ]; then make test; fi
+
+#----------------------------------------------------------------------------------------------
 FROM redis
 
 RUN set -e; apt-get -qq update; apt-get install -y libgomp1
 
-RUN set -ex;\
-    mkdir -p /usr/lib/redis/modules/;
+RUN mkdir -p /usr/lib/redis/modules/
 
-COPY --from=builder /redisai/install/ /usr/lib/redis/modules/
+COPY --from=builder /redisai/install-cpu/ /usr/lib/redis/modules/
 
 WORKDIR /data
 EXPOSE 6379
-CMD ["--loadmodule", "/usr/lib/redis/modules/redisai.so"]
+CMD ["--loadmodule", "/usr/lib/redis/modules/redisai.so", \
```

Comment:
Now that the default backend has landed, we may want to avoid hard-coding the backends here. What do you think?

Reply:
Already fixed but not pushed for some reason. Will do.